src/go/rpk/pkg/config/params.go

```diff
@@ -270,6 +271,43 @@ func (p *Params) Load(fs afero.Fs) (*Config, error) {
 	return c, nil
 }
 
+// Write writes loaded configuration parameters to redpanda.yaml.
+func (c *Config) Write(fs afero.Fs) error {
+	// It is the caller's responsibility to call write after loading the config file,
+	// this ensures that we don't write without default values, however,
+	// Writes depends on c.Config file so we better err here instead of panic in runtime.
+	if c.ConfigFile == "" {
+		return fmt.Errorf("config file not loaded, please load config file before writing to it")
+	}
+
+	b, err := yaml.Marshal(c)
+	if err != nil {
+		return fmt.Errorf("marshal error in loaded config, err: %s", err)
+	}
+
+	// Create a temp file.
+	layout := "20060102150405" // year-month-day-hour-min-sec
+	bFilename := "redpanda-" + time.Now().Format(layout) + ".yaml"
+	temp := filepath.Join(filepath.Dir(c.ConfigFile), bFilename)
+
+	_, err = fs.Create(temp)
```

Comment:
`afero.TempFile` creates the temp file without group permissions and Write might be called within redpanda group.

Reply:
So it looks like `fs.Create` uses `os.Create`, and under the hood, this will be
```go
return OpenFile(name, O_RDWR|O_CREATE|O_TRUNC, 0666)
```
you bring up a good point about perms -- I think we need to create this new file with the same perms as the existing file, as well as in the same group (chown below). We might need to create it with write perms to start, then write the file, drop the write perms, chown the file, and then finally move the file.

Reply:
Sure, this will also help with https://github.com/redpanda-data/redpanda/issues/4046